test/higher_order_ops/test_with_effects.py

```diff
@@ -662,6 +680,241 @@ def forward(self, arg0_1, arg1_1):
     return (getitem, mul, mul)""",
         )
 
+    @skipIfNoDynamoSupport
+    def test_effectful_op_in_backward(self):
+        with torch.library._scoped_library("_mylib", "FRAGMENT") as lib:
+            lib.define("foo(Tensor x) -> Tensor")
+
+            def foo_impl(a):
+                return a.clone()
+
+            def foo_bwd(ctx, grad):
+                return torch.ops._mylib.foo(grad)
+
+            for backend in ["CPU", "CUDA", "Meta"]:
+                lib.impl("foo", foo_impl, backend)
+
+            if torch._C._dispatch_has_kernel_for_dispatch_key(
+                "_mylib::foo", "Autograd"
+            ):
+                self.skipTest(
+                    "Double registration of Autograd kernel for test custom op"
+                )
+
+            torch.library.register_autograd("_mylib::foo", foo_bwd)
+
+            from torch._higher_order_ops.effects import (
+                _deregister_effectful_op,
+                _EffectType,
+                _register_effectful_op,
+            )
+
+            _register_effectful_op(torch.ops._mylib.foo.default, _EffectType.ORDERED)
```

Comment:
the SIDE_EFFECTS dict should probably be a WeakKeyDictionary -- the operator can go out of scope

Reply:
Will add TODO to SIDE_EFFECTS